app/src/androidTest/java/com/android/sample/ui/authentication/SignInScreenTest.kt

```diff
@@ -0,0 +1,113 @@
+    navHostController = mock(NavHostController::class.java)
+    navigationActions = NavigationActions(navHostController)
+  }
+
+  @Test
+  fun testEmailTextFieldExists() {
+    // Set the SignInScreen as the content to test
+    composeTestRule.setContent {
+      SignInScreen(
+          navigationActions = NavigationActions(navHostController)) // Passing null for simplicity
+    }
+
+    // Check if the email text field exists and is displayed
+    composeTestRule.onNodeWithTag("EmailTextField").assertIsDisplayed()
+  }
+
+  @Test
+  fun testPasswordTextFieldExists() {
+    // Set the SignInScreen as the content to test
+    composeTestRule.setContent {
+      SignInScreen(
+          navigationActions = NavigationActions(navHostController)) // Passing null for simplicity
+    }
+
+    // Check if the password text field exists and is displayed
+    composeTestRule.onNodeWithTag("PasswordTextField").assertIsDisplayed()
+  }
+
+  @Test
+  fun testSignInButtonExists() {
+    // Set the SignInScreen as the content to test
+    composeTestRule.setContent {
+      SignInScreen(
+          navigationActions = NavigationActions(navHostController)) // Passing null for simplicity
+    }
+
+    // Check if the sign-in button exists and is displayed
+    composeTestRule.onNodeWithTag("SignInButton").assertIsDisplayed()
+  }
+
+  @Test
+  fun testGoogleSignInButtonExists() {
+    // Set the SignInScreen as the content to test
+    composeTestRule.setContent {
+      SignInScreen(
+          navigationActions = NavigationActions(navHostController)) // Passing null for simplicity
+    }
+
+    // Check if the Google sign-in button exists and is displayed
+    composeTestRule.onNodeWithTag("GoogleSignInButton").assertIsDisplayed()
+  }
+
+  @Test
+  fun testAppLogoExists() {
+    // Set the SignInScreen as the content to test
+    composeTestRule.setContent {
+      SignInScreen(
+          navigationActions = NavigationActions(navHostController)) // Passing null for simplicity
+    }
+
+    // Check if the app logo exists and is displayed
+    composeTestRule.onNodeWithTag("AppLogo").assertIsDisplayed()
+  }
+
+  @Test
+  fun testContinueAsGuestButtonExists() {
+    // Set the SignInScreen as the content to test
+    composeTestRule.setContent {
+      SignInScreen(
+          navigationActions = NavigationActions(navHostController)) // Passing null for simplicity
+    }
+
+    // Check if the continue as guest button exists and is displayed
+    composeTestRule.onNodeWithTag("ContinueAsGuestButton").assertIsDisplayed()
+  }
+
+  @Test
+  fun testGoToSignUpButtonExists() {
+    // Set the SignInScreen as the content to test
+    composeTestRule.setContent {
+      SignInScreen(
+          navigationActions = NavigationActions(navHostController)) // Passing null for simplicity
+    }
+    // Check if the go to sign up button exists and is displayed
+    composeTestRule.onNodeWithTag("GoToSignUpButton").assertIsDisplayed()
+  }
+}
```

Comment:
Why did you divide the existence of the elements into multiple tests ? in the bootcamp we had done these all in one test